JavaScript-Runtime-(JSRT)-Overview.md

```diff
@@ -85,9 +85,117 @@ All other supported ES2015 features do not require any setup to run in JSRT.
 
 Features based on promises, such as [ES2016/ES7 Async Functions](https://blogs.windows.com/msedgedev/2015/09/30/asynchronous-code-gets-easier-with-es2016-async-function-support-in-chakra-and-microsoft-edge/), would require setting up promises before usage. 
 
+## Modules
+ES6 Modules are supported in ChakraCore. However, the host needs to specify how to resolve a module specifier since this part is host-specific not runtime-related. JSRT provides experimental APIs to set up ES6 modules. This [PR](https://github.com/Microsoft/ChakraCore/commit/50767f9f33812b4a036794a6a98404145563409f) gives you an overview of the related APIs and an example implementation (see WScriptJsrt.cpp) in the ch test host. Better documentation will be provided in the future.
+
 ## Experimental Features
 Experimental JavaScript features that are not yet stable are hidden behind an experimental flag. You can enable these features on a runtime by specifying **JsRuntimeAttributeEnableExperimentalFeatures** flag when **JsCreateRuntime** is called. 
 
+## Basic tasks
+
+This section aims to illustrate how to achieve a few common tasks with JSRT with code samples.
+
+### Expose native functions/objects to JavaScript
+
+The basic idea is to create a native function or (external) object, and expose it to JS by setting it as a property of an object JS is already aware of. For example, let's expose `console.log`, which many JavaScript hosts expose but isn't part of JavaScript.
+
+```cpp
+// a native function for logging
+JsValueRef CALLBACK LogCB(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+	for (unsigned int index = 1; index < argumentCount; index++)
+	{
+		if (index > 1)
+		{
+			wprintf(L" ");
+		}
+		JsValueRef stringValue;
+		JsConvertValueToString(arguments[index], &stringValue);
+		const wchar_t *string;
+		size_t length;
+		JsStringToPointer(stringValue, &string, &length);
+		wprintf(L"%s", string);
+	}
+	wprintf(L"\n");
+	return JS_INVALID_REFERENCE;
+}
+
+JsValueRef console, logFunc, global;
+JsPropertyIdRef consolePropId, logPropId;
+// create console object, log function, and set log function as property of console
+JsCreateObject(&console);
+JsCreateFunction(LogCB, nullptr, &logFunc);
+JsGetPropertyIdFromName(L"log", &logPropId);
+JsSetProperty(console, logPropId, logFunc, true);
+// set console as property of global object
+JsGetGlobalObject(&global);
+JsGetPropertyIdFromName(L"console", &consolePropId);
+JsSetProperty(global, consolePropId, console, true);
+```
+
+Now in JavaScript, you can do,
+```js
+console.log('Hello world');
+```
+
+If you need to expose an object that hosts data not directly exposed to JS, you can create an external object via `JsCreateExternalObject`.
```

Comment:
Should we show an example of this?

Reply:
I feel like the API shape is pretty self-explanatory so I left that out.

Reply:
I'm not sure that this line is helpful without an example at least illustrating what type of data you might expose and why you might expose it.

Reply:
Please also consider that we have `JsCreateExternalObjectWithPrototype` . IMHO we may show an example of that one instead (usage is available under test/native-tests)

Reply:
Didn't know that's a thing! Will look into it.